mission8/tests/auth.test.ts

```diff
@@ -0,0 +1,26 @@
+import request from 'supertest';
+
+import app from '../src/app';
+import prisma from '../src/utils/prisma';
+
+beforeAll(async () => {
+  await prisma.user.deleteMany();
+  await prisma.product.deleteMany();
+});
+
+test('회원가입 성공', async () => {
+  const res = await request(app).post('/auth/register').send({
+    email: 'user@test.com',
+    password: 'password',
+    username: 'test',
+  });
+  expect(res.status).toBe(201);
+});
+
+test('로그인 성공', async () => {
+  const res = await request(app).post('/auth/login').send({
+    email: 'user@test.com',
+    password: 'password',
+  });
+  expect(res.status).toBe(200);
+});
```

Comment:
특정 엔드포인트의 통합 테스트는, 보통 응답 페이로드까지 검증하시는 게 좋습니다.
실제 응답 데이터가 변경되더라도 해당 테스트는 통과하는데, 실제 프로덕션에서는 응답 자체가 변경되어서 클라이언트에 문제가 발생하는 경우도 많기 떄문이에요.


```suggestion
test('로그인 성공', async () => {
  const res = await request(app).post('/auth/login').send({
    email: 'user@test.com',
    password: 'password',
  });
  expect(res.status).toBe(200);
  expect(res.body).toHaveProperty('accessToken');
  expect(typeof res.body.accessToken).toBe('string');
  expect(res.body.accessToken.length).toBeGreaterThan(0);
});
```